### PR TITLE
Enhanced Security Information via Updating SECURITY.md

### PR DESCRIPTION
**Pull request by tvermaashutosh**
### What change does this PR introduce?

Enhanced/improved the Security Report Guidelines as well as the Security Contact Information via updating the SECURITY.md file. Making it more clear/comprehensible for the users.

### Why was this change needed?

Closes: #4665

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -1,43 +1,65 @@
-# Security Contact
+# Security
 
 **Contact:** security@novu.co
 
-At Novu, we prioritize the security of our systems. While we strive to make our systems as secure as possible, vulnerabilities can still exist. If you discover a vulnerability, we kindly request your assistance in helping us enhance our security measures and protect our clients.
+Based on [https://supabase.com/.well-known/security.txt](https://supabase.com/.well-known/security.txt)
```
Comment (Cliftonz) [inline SECURITY.md:5]: Please remove this line.

**Review by Cliftonz — CHANGES_REQUESTED** [review]

File: SECURITY.md (hunk 1 above)
Comment (tvermaashutosh) [reply SECURITY.md]: @Cliftonz 
I have the required changes as per your request.
Please merge the PR now.

File: SECURITY.md [hunk 2]
```diff
@@ -1,43 +1,63 @@
...
+
+## What *We* Promise
+
+1. We'll get back to you within 3 business days with our assessment of the report and an estimated date when we expect to resolve it.
+
+2. We will not take any legal action against you related to the report, if you have adhered to the reporting instructions above.
+
+3. We'll treat your report with utmost confidentiality and won't share your personal information with third parties without your consent.
+
+4. We'll be keping you updated of the progress toward fixing the issue.
```
Comment (Cliftonz) [inline SECURITY.md:57]: keping should be keeping

**Review by Cliftonz — CHANGES_REQUESTED** [review]

File: SECURITY.md (hunk 2 above)
Comment (tvermaashutosh) [reply SECURITY.md]: @Cliftonz 
Resolved the spelling mistake.

Comment (tvermaashutosh) [reply SECURITY.md]: Is it fine now?

**Review by Cliftonz — APPROVED** [review]

**Pull request merged**